### PR TITLE
change ZKP section

### PR DESCRIPTION
**Pull request by brentzundel**
This PR:

- Adds a version of the note initially drafter by Orie in #1026 
- modernizes and generalizes the language in the section in an attempt to make it more applicable to various zero knowledge proof mechanisms.
- adds a note that the examples will either be re-worked or removed.

I believe this PR prepares this section for for one of these possibilities:
1. The WG normatively specifies a method for securing VCs that supports ZKPs, in which case the examples and normative language in this section could be easily updated.
2. The WG does not normatively specify a method for securing VCs that supports ZKPs, in which case the examples and normative language in this section would be removed, and the remaining language could be moved to the implementation guide.

Signed-off-by: Brent Zundel <brent.zundel@gmail.com>

***
<a href="https://pr-preview.s3.amazonaws.com/brentzundel/vc-data-model/pull/1030.html" title="Last updated on Mar 21, 2023, 6:46 PM UTC (5cbc6a4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1030/aeaa1ce...brentzundel:5cbc6a4.html" title="Last updated on Mar 21, 2023, 6:46 PM UTC (5cbc6a4)">Diff</a>

### CODE REVIEW

File: index.html
Comment (TallTed) [inline index.html:2924]:
```suggestion
The following are some capabilities that are compatible with
<a>verifiable credentials</a>, that are made possible by zero-knowledge
proof mechanisms:
```

File: index.html
Comment (TallTed) [inline index.html:2938]:
```suggestion
<b>Blinding of the signature value that is shared with a <a>verifier</a>.</b> This
```

File: index.html
Comment (TallTed) [inline index.html:2943]:
```suggestion
<b>Privacy preserving identification of the <a>holder</a> and/or <a>subject</a>.</b> This
```

File: index.html
Comment (TallTed) [inline index.html:2961]:
```suggestion
We note that in most instances, for a <a>holder</a> to make use of zero knowledge
```

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-02-08-vcwg#section3-2) on 2023-02-08

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 3.2. change ZKP section (pr vc-data-model#1030)

_See github pull request [vc-data-model#1030](https://github.com/w3c/vc-data-model/pull/1030)._




**Brent Zundel:** should be straightforward to adjust further or get rid of the section.  

**Manu Sporny:** orie suggests process in order. Doing that.  
… processing #1026, #1028 then need review on #1030.  

</details>

File: index.html
Comment (quartzjer) [inline index.html:2913]:
```suggestion
knowledge proof mechanisms when we transition to v2 of the data model to CR.
```

File: index.html
Comment (quartzjer) [inline index.html:2991]:
```suggestion
to secure a <a>verifiable credential</a> using normatively defined methods that
```

**Review by quartzjer — APPROVED** [review]

File: index.html
Comment (dlongley) [inline index.html:2925]:
```suggestion
<a>verifiable credentials</a>, made possible by zero-knowledge
proof mechanisms:
```

File: index.html
Comment (dlongley) [inline index.html:2931]:
```suggestion
<a>verifier</a> with precisely the information they request and nothing more.
```

**Review by dlongley — APPROVED** [review]
I remain a bit concerned about over promising what ZKP can realistically do in various use cases, but this text hasn't changed that much from what was originally approved to be in the VCDM 1.1 spec and there will be another revision at some point in the future before 2.0 is finished, so +1. Thanks for cleaning this up!

**Review by Sakurann — COMMENTED** [review]
Selective disclosure, derived claim and unlinkability are three distinct features. By ZKP do we mean a way to secure VCs that fulfill all three or at least one? This needs to be clear. In the former, I think it is highly unlikely we will have a mature way before VC-data-model v2 CR, if the latter, SD-JWT will fall under this umbrella, but then the section name should be renamed from "ZKP".

**Comment by brentzundel** [comment]
> Selective disclosure, derived claim and unlinkability are three distinct features. By ZKP do we mean a way to secure VCs that fulfill all three or at least one? This needs to be clear. In the former, I think it is highly unlikely we will have a mature way before VC-data-model v2 CR, if the latter, SD-JWT will fall under this umbrella, but then the section name should be renamed from "ZKP".

I agree that if SD-JWT becomes a normative part of the spec, then this section should be expanded to include it.

File: index.html
Comment (TallTed) [inline index.html:2914]: If this doesn't make sense to you, try reading both versions of the line aloud.
```suggestion
knowledge proof mechanisms when we transition v2 of the data model to CR.
```

File: index.html
Comment (TallTed) [inline index.html:2965]:
```suggestion
mechanisms with <a>verifiable credentials</a> requires the <a>issuer</a> to
have secured the <a>verifiable credential</a> in a manner that supports these
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: index.html
Comment (Sakurann) [inline index.html:2909]:
```suggestion
        <h3>Selective Disclosure and Unlinkability</h3>
```

**Review by Sakurann — CHANGES_REQUESTED** [review]
it should be clear that the properties being discussed are mainly Selective Disclosure and unlinkability (via derivatives and identifier clinding), and one mechanism does not have to achieve all three at the same time.

File: index.html
Comment (msporny) [inline index.html:2990]:
```suggestion
        <p class="issue" title="ZKP examples need to be updated">
```

**Review by msporny — APPROVED** [review]
Approving as long as minor nits are fixed.

File: index.html
Comment (brentzundel) [inline index.html:2967]:
```suggestion
capabilities.
        </p>
        <p>
It should also be noted that zero-knowledge proof mechanisms are not the only
means of providing the above-mentioned capabilities. 
        </p>
```

**Comment by brentzundel** [comment]
@Sakurann and @TallTed I have made changes to the PR to reflect your comments. Please re-review.

**Review by TallTed — APPROVED** [review]
Much better than it was.

**Comment by msporny** [comment]
Merging this is currently blocked by @Sakurann. I've requested an explicit re-review. @Sakurann, please re-review the changes that @brentzundel has made to this PR so we can either merge it, or make further changes to address your concerns.

**Review by tplooker — APPROVED** [review]

File: index.html
Comment (Sakurann) [inline index.html:2956]:
```suggestion
Not all capabilities are supported in each zero-knowledge proof mechanisms.
```

**Review by Sakurann — APPROVED** [review]
Despite the change in the title, the section is still all about "zero-knowledge proof", which is a specific term and would not include things like SD-JWT. But given there are caveats to rewrite/remove this section written in the text, will keep the reservations until when that PR comes in later.

File: index.html
Comment (TallTed) [reply index.html:2956]:
```suggestion
A given zero-knowledge proof mechanism might not support all capabilities described here.
```

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-03-14-vcwg-special#section2-2) on 2023-03-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.2. change ZKP section (pr vc-data-model#1030)

_See github pull request [vc-data-model#1030](https://github.com/w3c/vc-data-model/pull/1030)._




**Manu Sporny:** I think we're ready to merge this.  
… There are some changes that Kristina and Ted have suggested, but I think those are good to go unless someone here objects.  

> *Dave Longley:* +1 to merge with whatever editorial fixes.

**Brent Zundel:** ok, I'll apply language and merge.  

</details>

File: index.html
Comment (TallTed) [inline index.html:2925]:
```suggestion
Some capabilities that are compatible with <a>verifiable credentials</a> which
are made possible by zero-knowledge proof mechanisms:
```

File: index.html
Comment (TallTed) [inline index.html:2948]:
```suggestion
be shared. This capability can also be used to combine multiple
```

**Comment by David-Chadwick** [comment]
Have you considered the impact of these edits on section 5.4 (data schemas) as this has an example that concerns ZKPs. Are edits also needed to this section?

**Comment by Sebastian-Elfors-IDnow** [comment]
Thanks for the re-write of the ZKP section. It's a great improvement.

For completeness, the term "unlinkability" could be introduced. It is related to the bullet about "blinded signatures", which could be extended as follows: 

"Blinded signatures also allows for unlinkable proofs, which removes a common source of correlatable properties about a holder during multiple presentations to one or more verifiers."

**Pull request merged**
